sqlmesh/lsp/main.py

```diff
@@ -31,14 +32,85 @@ def __init__(
         """
         self.server = LanguageServer(server_name, version)
         self.context_class = context_class
-        self.context: t.Optional[Context] = None
+        # A tuple of (context, reverse_map) where the reverse_map is uri to model name
+        self.context_and_reverse_map: t.Optional[t.Tuple[Context, t.Dict[str, str]]] = None
```

Comment:
Any reason this shouldn't be 2 variables?

Reply:
It's not pretty in this world, but for me, the reason to do this is that you accurately encode the states that a state can be: 
1. loaded with both Context and the dict
2. not loaded 

Rather than allowing the inconsistent states that sit in between.